src/NLog/Common/InternalLogger.cs

```diff
@@ -198,7 +203,7 @@ public static void Log(Exception ex, LogLevel level, [Localizable(false)] string
         /// <param name="args">optional args for <paramref name="message"/></param>
         private static void Write([CanBeNull]Exception ex, LogLevel level, string message, [CanBeNull]object[] args)
         {
-            if (ex != null && ex.MustBeRethrownImmediately())
+            if (IsSeriousException(ex))
```

Comment:
I think this is a bit overkill, a new method, as it used only once and it's only 2 statements